app/src/pages/inside/testExecutionsPage/testExecutionsFilters/launchAttributesFilter/launchAttributesFilter.jsx

```diff
@@ -0,0 +1,167 @@
+import { EditableAttribute } from 'componentLibrary/attributeList/editableAttribute';
+import { COMMON_LOCALE_KEYS } from 'common/constants/localization';
+import { FILTER_KEYS } from '../constants';
+import { getStoredFilter, setStoredFilter } from '../utils';
+import styles from './launchAttributesFilter.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  label: {
+    id: 'TestExecutionsPage.filters.attributesLabel',
+    defaultMessage: 'Launch attributes',
+  },
+  addAttributeButton: {
+    id: 'TestExecutionsPage.filters.addAttributeButton',
+    defaultMessage: 'Add attribute',
+  },
+  keyPlaceholder: {
+    id: 'TestExecutionsPage.filters.keyPlaceholder',
+    defaultMessage: 'Specify key',
+  },
+  valuePlaceholder: {
+    id: 'TestExecutionsPage.filters.valuePlaceholder',
+    defaultMessage: 'Specify value',
+  },
+});
+
+const LaunchAttributesFilterComponent = ({ filter, onFilterChange }) => {
+  const { formatMessage } = useIntl();
+  const activeProject = useSelector(activeProjectSelector);
+  const userId = useSelector(idSelector);
+  const [isOpened, setIsOpened] = useState(false);
+  const [attributes, setAttributes] = useState([]);
+
+  const getURIKey = URLS.launchAttributeKeysSearch(activeProject);
+  const getURIValue = (key) => URLS.launchAttributeValuesSearch(activeProject, key);
+
+  useEffect(() => {
+    if (filter) {
+      const parsedAttributes = parseQueryAttributes({ value: filter });
+      setAttributes(parsedAttributes);
+    } else {
+      setAttributes([]);
+    }
+  }, [filter]);
+
+  useEffect(() => {
+    if (userId && activeProject) {
+      const stored = getStoredFilter(userId, activeProject, FILTER_KEYS.ATTRIBUTES);
+      if (stored && !filter) {
+        onFilterChange(stored);
+      }
+    }
+  }, [filter, onFilterChange, userId, activeProject]);
+
+  const handleAddAttribute = () => {
+    setIsOpened(true);
+  };
+
+  const handleChange = (newAttributes) => {
+    setAttributes(newAttributes);
+    const formattedAttributes = newAttributes.map(formatAttribute).join(',');
+    if (userId && activeProject) {
+      setStoredFilter(userId, activeProject, FILTER_KEYS.ATTRIBUTES, formattedAttributes);
+    }
+    onFilterChange(formattedAttributes);
+  };
+
+  const handleConfirm = (attribute) => {
+    const updatedAttributes = [...attributes, attribute];
+    handleChange(updatedAttributes);
+    setIsOpened(false);
+  };
+
+  const handleCancel = () => {
+    setIsOpened(false);
+  };
+
+  const handleRemove = (allAttributes, index, onChange) => () => {
+    const newAttributes = [...allAttributes];
+    newAttributes.splice(index, 1);
+    onChange(newAttributes);
+  };
+
+  return (
+    <div>
+      <FieldLabel>{formatMessage(messages.label)}</FieldLabel>
+      <div className={cx('attributes-list')}>
+        {attributes.length > 0 &&
+          attributes.map((attribute, index) => (
```

Comment:
I suppose the array length check is redundant here as the render will be skipped anyway in case of empty array.